source/Calamari/ArgoCD/Commands/CommitToGitCommand.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Calamari.ArgoCD.Conventions;
+using Calamari.Commands;
+using Calamari.Commands.Support;
+using Calamari.Common.Commands;
+using Calamari.Common.Features.Behaviours;
+using Calamari.Common.Features.Packages;
+using Calamari.Common.Features.Substitutions;
+using Calamari.Common.Plumbing.Deployment;
+using Calamari.Common.Plumbing.FileSystem;
+using Calamari.Common.Plumbing.Logging;
+using Calamari.Common.Plumbing.Variables;
+using Calamari.Deployment;
+using Calamari.Deployment.Conventions;
+
+namespace Calamari.ArgoCD.Commands
+{
+    [Command(Name, Description = "Write populated templates from a package into one or more git repositories")]
+    public class CommitToGitCommand : Command
+    {
+        public const string PackageDirectoryName = "package";
+
+        public const string Name = "commit-to-git";
+
+        readonly ILog log;
+        readonly IVariables variables;
+        readonly ICalamariFileSystem fileSystem;
+        readonly IExtractPackage extractPackage;
+        readonly INonSensitiveSubstituteInFiles substituteInFiles;
+        PathToPackage pathToPackage;
+
+        public CommitToGitCommand(
+            ILog log,
+            IVariables variables,
+            ICalamariFileSystem fileSystem,
+            IExtractPackage extractPackage,
+            INonSensitiveSubstituteInFiles substituteInFiles)
+        {
+            this.log = log;
+            this.variables = variables;
+            this.fileSystem = fileSystem;
+            this.extractPackage = extractPackage;
+            this.substituteInFiles = substituteInFiles;
+
+            Options.Add("package=",
+                        "Path to the NuGet package to install.",
+                        v => pathToPackage = new PathToPackage(Path.GetFullPath(v)));
+        }
+
+        public override int Execute(string[] commandLineArguments)
+        {
+            Options.Parse(commandLineArguments);
+
+            var conventions = new List<IConvention>
+            {
+                new DelegateInstallConvention(d =>
+                                              {
+                                                  var workingDirectory = d.CurrentDirectory;
+                                                  var packageDirectory = Path.Combine(workingDirectory, PackageDirectoryName);
+                                                  fileSystem.EnsureDirectoryExists(packageDirectory);
+                                                  extractPackage.ExtractToCustomDirectory(pathToPackage, packageDirectory);
+                                              }),
+                new SubstituteInFilesConvention(new NonSensitiveSubstituteInFilesBehaviour(substituteInFiles, PackageDirectoryName)),
```

Comment:
not for this pr, but should we have a `NonSensitiveSubstituteInFilesConvention`?

Reply:
I wonder if the heirarchy didn't add value - and we should have passed in a boolean to say "include SensitiveValues" .... lots of empty files do do not much :/